php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -1403,12 +1403,23 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     function getFieldValue(string $field)
     {
-        $allValues = $this->getInstanceData();
-
-        if (!array_key_exists($field, $allValues)) {
-            throw new \OutOfBoundsException("Invalid field for instrument");
+        $allValues  = $this->getInstanceData();
+        $dictionary = self::factory(
+            $this->loris,
+            $this->testName
+        )->getDataDictionary();
```

Comment:
@driusan `getDataDictionary()` modifies the `$this->page` value altering the active page of the instrument causing the weird page change on save.

I fixed it here simply by getting the data dictionary from another instance of the same instrument... I don't like the fact that getDataDictionary modifies the instrument on which it is called however. To be discussed